test/graph_test.cpp

```diff
@@ -341,3 +342,117 @@ TEST(GraphTest, MultipleInsertThreads) {
   g.set_verifier(std::make_unique<FileGraphVerifier>("./cumul_sample.txt"));
   g.connected_components();
 }
+
+TEST(GraphTest, MTStreamWithMultipleQueries) {
+  for(int i = 1; i <= 10; i++) {
+    GraphConfiguration config;
+    config.gutter_sys = STANDALONE;
+
+    const std::string fname = __FILE__;
+    size_t pos = fname.find_last_of("\\/");
+    const std::string curr_dir = (std::string::npos == pos) ? "" : fname.substr(0, pos);
+    BinaryGraphStream_MT stream(curr_dir + "/res/multiples_graph_1024.data", 256);
```

Comment:
consider doing this test with a random graph or graph with deletions

Reply:
We currently have no ability to define random binary streams. However, it wouldn't be hard to add some deletions to the .data graph stream

Reply:
I'll add that as an issue